src/elements/components/download-button-edit-popover.tsx

```diff
@@ -0,0 +1,158 @@
+import { Popover, Transition } from '@headlessui/react';
+import { Fragment, useState } from 'react';
+import { Resource } from '../../lib/schema';
+
+export interface EditResourceProps {
+    resource?: Resource;
+    onChange?: (value: Resource) => void;
```

Comment:
This is never `undefined` in practice. Since the whole purpose of this is component is to edit something, I would argue that it doesn't make sense of `onChange` to be `undefined`.